src/gui/update.rs

```diff
@@ -263,19 +278,39 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
                 Message::None,
             ));
         }
-        Message::UpdateWowDirectory(chosen_path) => {
-            log::debug!("Message::UpdateWowDirectory(Chosen({:?}))", &chosen_path);
-            let path = wow_path_resolution(chosen_path);
-            log::debug!("Message::UpdateWowDirectory(Resolution({:?}))", &path);
+        Message::UpdateWowDirectory((chosen_path, flavor)) => {
+            log::debug!(
+                "Message::UpdateWowDirectory(Chosen({:?} - {:?}))",
+                &chosen_path,
+                &flavor
+            );
+            if let Some(path) = wow_path_resolution(chosen_path) {
+                log::debug!("Message::UpdateWowDirectory(Resolution({:?}))", &path);
+
+                if let Some(flavor) = flavor {
+                    // If a flavor is supplied we only update path for that specific flavor.
+                    let flavor_path = ajour.config.get_flavor_directory_for_flavor(&flavor, &path);
+                    if flavor_path.exists() {
+                        ajour.config.wow.directories.insert(flavor, flavor_path);
+                    }
+                } else {
+                    // If no flavor is supplied it will find as many flavors as possible in the path.
```

Comment:
If I understand correctly, the only time `None` is passed is when the user has NO directories selected (first time launching Ajour)? Otherwise, it always for a single flavor? Smart.

Reply:
Yeah exactly - I think it makes the most sense that we do a "find all flavors initially from this folder" and from that point we only work on flavor base.